sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -282,9 +282,44 @@ pi_result piDeviceGetInfo(pi_device device, pi_device_info paramName,
     // For details about Intel UUID extension, see
     // sycl/doc/extensions/supported/sycl_ext_intel_device_info.md
   case PI_DEVICE_INFO_UUID:
-  case PI_DEVICE_INFO_ATOMIC_MEMORY_ORDER_CAPABILITIES:
   case PI_DEVICE_INFO_ATOMIC_MEMORY_SCOPE_CAPABILITIES:
     return PI_ERROR_INVALID_VALUE;
+  case PI_DEVICE_INFO_ATOMIC_MEMORY_ORDER_CAPABILITIES: {
+    // This query is missing beore OpenCL 3.0
+    // Check version and handle appropriately
+    OCLV::OpenCLVersion devVer, platVer;
+    cl_platform_id platform;
+    cl_device_id deviceID = cast<cl_device_id>(device);
+
+    auto ret_err = clGetDeviceInfo(deviceID, CL_DEVICE_PLATFORM,
+                                   sizeof(cl_platform_id), &platform, nullptr);
+    if (ret_err != CL_SUCCESS) {
+      return cast<pi_result>(ret_err);
+    }
+
+    ret_err = getDeviceVersion(deviceID, devVer);
+    if (ret_err != CL_SUCCESS) {
+      return cast<pi_result>(ret_err);
+    }
+
+    ret_err = getPlatformVersion(platform, platVer);
+    if (ret_err != CL_SUCCESS) {
+      return cast<pi_result>(ret_err);
+    }
+
+    if (platVer < OCLV::V3_0 || devVer < OCLV::V3_0) {
+      setErrorMessage(
+          "OpenCL version for device and/or platform is less than 3.0",
+          PI_ERROR_INVALID_OPERATION);
+      return PI_ERROR_INVALID_OPERATION;
+    }
+
+    // Guaranteed to return at least relaxed memory order
+    cl_int result = 1;
+    // TODO: Check for support for the rest of the capabilities
+    std::memcpy(paramValue, &result, sizeof(cl_int));
+    return PI_SUCCESS;
```

Comment:
We don't do any query at all, why do we perform versions check then? Please note that in SYCL this query should return a result regardless of the underlying backend.

I suggest that we refactor the logic here:
```
if (OpenCLVer < 3.0) {
  return default; // In accordance with SYCL 2020, default is "relaxed" if I remember correctly
}

doOpenCLQuery();
convertOpenCLResultIntoSYCLResult();
return result;
```

Please also note that [`CL_DEVICE_ATOMIC_MEMORY_CAPABILITIES`](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_DEVICE_ATOMIC_MEMORY_CAPABILITIES) returns mask for both `order` and `scope` and we need to apply some post-processing to extract the part we need and convert it to SYCL enum values